content/en/core/overview/self-hosting-multiple-nodes.md

```diff
@@ -0,0 +1,285 @@
+
+### What are the four nodes used for?
+* Node 1 - CHT-core - runs the core functionality of the CHT like API, sentinel
+* Node 2, 3 & 4 - Couchdb Nodes (A 3 node couchdb cluster)
+
+### Networking Setup
+
+Make sure the following ports are open for the nodes:
+* `7946 TCP` - For docker communication amongst nodes
+* `7946 UDP` - For docker communication amongst nodes
+* `2377 TCP` - Docker cluster management communication
+* `4789 UDP` - Docker overlay network traffic
+* `ICMP` - For ping
+
+**Security Note**: As a security measure, be sure to restrict the IP addresses of the four nodes only to be able to connect to these ports.
+
+## Setup Docker Swarm and Create an Overlay Network
+
+### Node 1 - Main node
+
+1. Initialize swarm mode.
+
+`sudo docker swarm init`
+
+You should get the output:
+
+```
+Swarm initialized: current node (some-long-id) is now a manager.
+To add a worker to this swarm, run the following command:
+
+    docker swarm join --token <very-long-token-value> <main-server-private-ip>:2377
+
+To add a manager to this swarm, run 'docker swarm join-token manager' and follow the instructions.
+
+```
+
+2. Create overlay network
+
+`sudo docker network create --driver=overlay --attachable cht-overlay`
+
+### Node2, Node3 & Node4
+
+On each of these nodes run the join command given to you above.
+
+`docker swarm join --token <very-long-token-value> <main-server-private-ip>:2377`
+
+
+## Configuration - Node 1 - CHT-core
+
+### Setup Docker
+
+1. `sudo apt update`
+2. `sudo apt install docker.io`
+3. `sudo apt install docker-compose`
+
+### Directory Structure
+
+Create the following directory structure. This is important.
+
+```
+|--CHT
+    |__ compose/
+    |__ certs/
+    |__ upgrade-service/
+```
+
+Explanations:
+
+* compose - contains the docker-compose files for cht-core and couchdb.
+* certs - the path that you'll put SSL certificates.
+* upgrade-service - the path for the docker-compose file of the upgrade-service.
+
+### Download required docker-compose files
+
+The following commands download the 4.0.1 version. If you want a different version, you would have to change `medic:4.0.1` to the version you'd like to install.
+
+1. Download the cht-core docker-compose file. Run the following command from the `CHT` directory above.
+`curl -s -o ./compose/cht-core.yml https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:4.0.1/docker-compose/cht-core.yml`
+2. Download the upgrade service docker-compose file.
+`curl -s -o ./upgrade-service/docker-compose.yml https://raw.githubusercontent.com/medic/cht-upgrade-service/main/docker-compose.yml`
+
+### Docker-compose modifications
+
+1. Go to the networks section and add the following:
+
+```
+  cht-overlay:
+    driver: overlay
+    external: true
+```
+
+2. Go to each of the containers and add the following:
+
+```
+    networks:
+      - cht-overlay
+```
```

Comment:
fix indentation (btw - GH isn't letting me put triple ticks  ```  in side suggestions so I'm using indentations :frowning_face:  ) and clarify which files we're editing


Edit both the `cht-core.yml` and `docker-compose.yml` compose files:

```suggestion
1. At the end of each compose file, ensure the 2 `networks:` sections (1 in `cht-core.yml` and 1 in `docker-compose.yml`) look like this by adding three lines:


        networks:
            cht-net:
               name: ${CHT_NETWORK:-cht-net}
            cht-overlay:
               driver: overlay
               external: true


2. For all 6 services in the compose files (5 in `cht-core.yml` and 1 in `docker-compose.yml`), update the `networks:` section to look like this:

        networks:
          - cht-overlay
          - cht-net

```